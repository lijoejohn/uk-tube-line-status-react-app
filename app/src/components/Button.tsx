type ButtonProps = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;
function Button(props: ButtonProps): JSX.Element {
  const { children, ...otherProps } = props;
  return <button {...otherProps}>{children}</button>;
}

export default Button;
