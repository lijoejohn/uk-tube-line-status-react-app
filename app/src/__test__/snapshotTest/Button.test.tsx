import * as renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button Component', () => {
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Button className="govuk-button" type="button">
          Test
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
