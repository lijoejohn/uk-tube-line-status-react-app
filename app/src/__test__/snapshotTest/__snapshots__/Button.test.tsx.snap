// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match the snapshot 1`] = `
<button
  className="govuk-button"
  type="button"
>
  Test
</button>
`;
